app/controllers/articles_controller.rb

```diff
@@ -0,0 +1,52 @@
+class ArticlesController < ApplicationController
+  
+  http_basic_authenticate_with name: "mayank", password: "bv-mayank", except: [:index, :show]
+
+  def index
+    @articles = Article.all
```

Comment:
Never write code like this.
It may work where data is always too small but in all other cases this will be a bad way to do it.
Ideally here we should use pagination but even if pagination is not there, a limit can be used.
So, a slightly better way to do this would be `Article.all..order(:id => :desc)limit(100)`